omnipaxos/src/messages.rs

```diff
@@ -73,6 +73,8 @@ pub mod sequence_paxos {
         pub sync_idx: u64,
```

Comment:
To avoid confusion I think we should replace the `decided_snapshot, suffix, sync_idx` fields with the following:
```
enum SyncItem {
        Entries(u64, Vec<T>), // sync_idx + suffix (receiver applies suffix at sync_idx)
        SnapshotAndEntries(SnapshotType<T>, Vec<T>) // receiver applies snapshot (with decided_idx) and appends suffix)
}

```

Reply:
I think of the AcceptSync as 3 separate steps (adopt Snapshot, adopt suffix, then adopt StopSign). Trying to keep these updates logically separated will make it easier to define the atomic transaction in the future.

Reply:
Yes, but I think this SyncItem would help with that. Because then when matching the different variants we would potentially rollback what is required. 

I don't like that we send a compacted index (that is 0) in the cases we don't even send a snapshot, and similarly a sync index when we use snapshots.